handlers/response/response.go

```diff
@@ -22,32 +23,60 @@ type sgtBaseResponse struct {
 // storing custom endpoint response values
 type SGTCustomResponse map[string]interface{}
 
-func getResponseJSON(response interface{}) []byte {
+func writeResponseJSON(respWriter http.ResponseWriter, response interface{}) bool {
 
 	respJSON, err := json.Marshal(response)
 	if err != nil {
-		// If there is an error marshaling the interface, return a basic error
+		//If there is an error marshaling the interface, write a basic error
 		errString := fmt.Sprintf("response failed to marshal to json: %s", err)
-		newResp := sgtBaseResponse{Message: errString, Status: errString}
-		respJSON, _ = json.Marshal(newResp)
+		http.Error(respWriter, errString, http.StatusInternalServerError)
+		return false
 	}
-	return respJSON
+
+	return writeResponse(respWriter, string(respJSON))
+}
+
+func writeResponse(respWriter http.ResponseWriter, value string) bool {
+
+	// Write the response to the http.ResponseWriter using io.WriteString
+	if _, err := io.WriteString(respWriter, value); err != nil {
+		//If there is an error writing the repsonse, write an error
+		errString := fmt.Sprintf("failed to write response: %s", err)
+		http.Error(respWriter, errString, http.StatusInternalServerError)
+		return false
+	}
+
+	return true
 }
 
 // WriteError will write the passed error to the http response writer
 func WriteError(respWriter http.ResponseWriter, errorString string) {
+	if !writeResponseJSON(respWriter, sgtBaseResponse{Message: errorString, Status: statusError}) {
+		return
+	}
+
+	// Set the header type on success
 	respWriter.Header().Set("Content-Type", "application/json")
-	respWriter.Write(getResponseJSON(sgtBaseResponse{Message: errorString, Status: statusError}))
 }
 
 // WriteSuccess will write the a success status and optional message to the http response writer
 func WriteSuccess(respWriter http.ResponseWriter, optionalMessage string) {
+	if !writeResponseJSON(respWriter, sgtBaseResponse{Message: optionalMessage, Status: statusSuccess}) {
+		return
+	}
+
+	// Set the header type/status on success
+	respWriter.WriteHeader(http.StatusOK)
 	respWriter.Header().Set("Content-Type", "application/json")
```

Comment:
@ryandeivert is there ever an instance when the content-type would not be json?  Generally we want to try to make everything a json response, but if there's a use-case for non-json let me know.  (the whole point being, why not set the header type once, no matter what)

Reply:
Purposely, probably not :) the only time (by design) I think the content type will be anything other than json is if `http.Error` is called - but that would overwrite the json content type if it existed already. I'll make the change 👍